### ISSUE TITLE
[SCB-0002] Command Duplication

### ISSUE THREAD

**Issue by N3v1** [ScribbleLabApp/ScribbleCareBear#2]
# [Bug]: Commands of Discord Bots are Duplicated

### Metadata

**Issue Number:** SCB-0002 #2
**Status:** Help needed · WIP

### Overview
I have encountered a recurring issue where the commands executed by my Discord bots are being duplicated on my server. This problem has been persistent and is affecting the functionality and user experience of the bot. Below are the details of the issue:

When a command is issued to the bot, it is processed twice, resulting in duplicated responses. This duplication affects all commands, regardless of their complexity or purpose. The issue is consistent and reproducible with each command execution.

## Steps to Reproduce

1. Invite the bot to a Discord server **[TESTING PURPOSES ONLY]** (better **join our [Discord Server](https://discord.gg/Y5mUJbNC8h)**)
2. Execute any command that the bot responds to.
3. Observe that the bot commands in the command panel appear twice

## Expected Behavior
The bot should only render/provide each command only once and send a single response per command execution.

## Actual Behavior
Discord renders/shows commands twice, leading to confusion and clutter in the command panel.

## Impact
Users see duplicated commands, which can be misleading and annoying. The bot's functionality appears unreliable, and users may lose trust in its responses. The server's chat becomes cluttered with repeated commands, disrupting the flow of conversation.

## Additional Information
The bot is developed using JavaScript and relies on the Discord.js library. The issue started occurring after a recent update to the bot's codebase. Other bots on the same server do not exhibit this duplication issue.

### Logs and Error Messages
Attached are the relevant logs and error messages observed during the issue. (Please attach any relevant logs or screenshots here)

### Request for Assistance
I would appreciate any guidance or suggestions on how to resolve this duplication issue. If there are any known solutions or similar issues reported, please point me in the right direction.

Thank you for your assistance in resolving this matter.
Have a great day! 🧡

---

### Logs

```md
[2024-11-02T15:33:45.975Z] Logged in as Scribble Care Bear#6474
[2024-11-02T15:33:45.989Z] [SCB]: Registering new commands...
[2024-11-02T15:33:46.238Z] [SCB]: Successfully registered application commands!
[2024-11-02T15:33:46.238Z] [SCB]: Registered commands: about, ping, resources, say
[2024-11-02T15:34:18.939Z] [SCB]: Executing command: say
[2024-11-02T15:34:24.732Z] [SCB]: Executing command: resources
[2024-11-02T15:34:30.692Z] [SCB]: Executing command: ping
[2024-11-02T15:34:37.764Z] [SCB]: Executing command: about
```

### Screenshots

![SCR-20241102-oxac](https://github.com/user-attachments/assets/7d7b4ee5-f0cb-4a84-8f40-65d4c4f08281)
img1: ScribbleCareBear Commands Panel (shot taken on [**ScribbleLabApp Discord Server**](https://discord.gg/Y5mUJbNC8h))

**Comment by N3v1** [comment]
### [Update]: 02112024_17:10 
Investigation showed that in a playground/testing server no commands were duplicated which results to the theory that we are registering both guild and global commands which results into duplicated commands

**Comment by N3v1** [comment]
### [Update]: 02112024_17:26

We added a `clearGlobalCommands(token,applicationID)` function that clears up our old global commands.

### Changes that were made
```js
async function clearGlobalCommands(token, applicationID) {
    const rest = new REST({ version: "10" }).setToken(token);

    try {
        await rest.put(
            Routes.applicationCommands(applicationID), 
            { body: [] }
        );

        log("[SCB]: Successfully cleared global commands!");
    } catch (e) {
        log(`[SCB]: Error clearing global commands: ${e}`);
    }
}
```
[commands.js]

```js
...
client.once(Events.ClientReady, async () => {
    log(`Logged in as ${client.user.tag}`);
    
    try {
        await clearGlobalCommands(token, applicationID);

        // Register commands once when ScribbleCareBear becomes ready
        const commands = await registerCommands(token, applicationID, guildID);
        client.commands.clear();        //< Populate the commands map
        commands.forEach((cmd) => client.commands.set(cmd.data.name, cmd));

        log(`[SCB]: Registered commands: ${Array.from(client.commands.keys()).join(", ")}`);
    } catch (error) {
        log(`[SCB]: Failed to register commands: ${error}`);
    }
});
...
```
[index.js]

**Comment by N3v1** [comment]
### [Update]: 02112024_18:40

Resolved in commit [60e298c](https://github.com/ScribbleLabApp/ScribbleCareBear/commit/60e298c1b340dd4cb68a4f0915e94137c57c4ef9)
